src/Message/Communicate.h

```diff
@@ -247,7 +247,7 @@ class Communicate : public CommunicatorTraits
 
 #ifdef HAVE_MPI
   /// mpi3 communicator wrapper
-  mpi3::communicator comm;
+  mutable mpi3::communicator comm;
```

Comment:
Is this just a temporal workaround?

Reply:
Seems to be permanent, indicating that `Communicate` is an explicitly non thread-safe class out of the box.